middleware/proxy/google.go

```diff
@@ -218,23 +219,30 @@ func newUpstream(hosts []string, old *staticUpstream) Upstream {
 			Conns:       0,
 			Fails:       0,
 			FailTimeout: upstream.FailTimeout,
-			Unhealthy:   false,
+			OkUntil:     time.Now().Add(time.Duration(24 * 365 * 100 * time.Hour)), // now + 100 years
```

Comment:
This is for the no-health-check case - in which case OkUntil has to always be after time.Now().  If healthchecks are configured, the first one will override this.  I needed to pick a value guaranteed to be later than Now() for all reasonable circumstances.  If there's a suitable constant, sure.